File: repetitions.cpp
/*problem statement:You are given a DNA sequence: a string consisting of characters A, C, G, and T. 
The task is to find the longest repetition in the sequence. This is a maximum-length substring containing only one type of character.*/
#include<bits/stdc++.h>
using namespace std;
int main(){
	string input;
	cin>>input;//taking string input
	int count=1,length=input.size(),max_length=1;//maxlength is our answer 
	if(length==0){
		cout<<"0"<<endl;//corner case to check if string is provided or not
		return 0;
	}
	else{
		for(int i=1;i<length;i++){
			if(input[i]==input[i-1]){
				count++;
				if(count>max_length) max_length=count;//updates max_length if larger substring of same characters are found.
			}
			else{
				count=1;
			}
		}
		cout<<max_length<<endl;
		return 0;
	}
}

File: Python/Repetitions.py
# Time limit: 1.00 s Memory limit: 512 MB
# You are given a DNA sequence: a string consisting of characters A, C, G, and T. Your task is to find the longest repetition in the sequence. This is a maximum-length substring containing only one type of character.
#
# Input
#
# The only input line contains a string of n characters.
#
# Output
#
# Print one integer: the length of the longest repetition.
#
# Constraints
# 1≤n≤106
# Example
#
# Input:
# ATTCGGGA
#
# Output:
# 3

max = 0

string = input()
i=0
while(i<len(string)):
    m=1
    for j in range(i+1,len(string)):
        if string[j] != string[i]:
            break
        else:
            m+=1
    if max<m:
        max = m
        i+=m
    i+=1

print(max)
